Automattic/jetpack
# Add Google Analytics module

Synopsis: Add the existing Google Analytics plugin from WordPress.com as a Jetpack module. It is intended to be enabled for Jetpack Professional users.

Responsible: @rodrigoi  

Technical approach: 

1. `wp-google-analytics` plugin distributed _as-is_ as a module.
2. `class.wpcom-json-api-site-settings-endpoint.php` was modified to include a _Jetpack or WordPress.com_ check when updating the `wga` option.
3. The JSON API was modified to support a `google_analytics_tracking_id` field. That is translated to the same `wga` option WordPress.com uses.
4. _Engagement_ settings page was modified to support paid plans detection, and to include the new Google Analytics settings form.

#### Testing instructions:

* Run this branch on a WordPress installation.
* A new _Google Analytics_ item should show on the _Engagement_ tab
* only sites with a _Professional_ plan should be able to activate the module. Otherwise, a _upgrade_ button should show.
* when active and configured, the Google Analytics script should be added to any page.

It'll be ideal to include this on `4.4`, but set as `4.4.1` given the short timeline for review.

Complete: 80%

Difficulty: 

Last Updated: 11/11 by @jeherve
